README.md

```diff
@@ -7,11 +7,60 @@
 
 ## Overview
 
-This lesson is a very high-level over view of the React component lifecycle. We
-want to walk students through the whole shebang.
+As you've learned previously, a React component has two sets of properties: **props** and **state**. Props are given to the component by it's parent. You can think of props as an external influence that the component has no control over - like the weather. The other set of properties the component has is called the state. A component's state is internal to the component and it can change when the props changing or the user interacting with the component. For example, if the weather went from sunny to raining, you would put on a raincoat and wellies. In React component terms, your state could go from "wearing a jacket" to "wearing a raincoat and wellies". You know to put on a raincoat when it's raining because you've been told by your parents. A React component would have to be told what to do by the programmer - that's you! :)
```

Comment:
The Facebook code (below for reference) is a good example of setting state using props because it's a computed values based on the _difference_ of the old and new props.

```js
componentWillReceiveProps: function(nextProps) {
  this.setState({
    likesIncreasing: nextProps.likeCount > this.props.likeCount
  });
}
```

In your example, you're basically only checking the new props - this stuff could be done directly in the `render()` method:

```js
render() {
  const whatToWear = this.props.weather === 'raining' ? 'jacket' : 'raincoat';

  return (
    <p>I should be wearing a {whatToWear} today.</p>
  );
}
```

Doing something like this in the Facebook example would be impossible, because at that point you only have access to the current props - necessitating the need for using state. If we translate the Facebook example to our current weather narrative, we could have an example where we check if the temperature has risen or dropped (or maybe even with how many degrees) and set the state accordingly.

I'm not sure if I'm making a whole lot of sense here - let me know if you still have any questions!